meteor/sofie-server/data-access/repositories/test/mongo-part-repository.spec.ts

```diff
@@ -0,0 +1,174 @@
+import { MongoPartRepository } from '../mongo/mongo-part-repository'
+import { Part, PartInterface } from '../../../model/entities/part'
+import { Db } from 'mongodb'
+import { MongoEntityConverter } from '../mongo/mongo-entity-converter'
+import { PartRepository } from '../interfaces/part-repository'
+import { MongoDatabase } from '../mongo/mongo-database'
+import { anyString, anything, instance, mock, verify, when } from 'ts-mockito'
+import { MongoTestDatabase } from './mongo-test-database'
+import { PieceRepository } from '../interfaces/piece-repository'
+import { Piece, PieceInterface } from '../../../model/entities/piece'
+import { DeletionFailedException } from '../../../model/exceptions/deletion-failed-exception'
+
+const COLLECTION_NAME = 'parts'
+
+describe(`${MongoPartRepository.name}`, () => {
+	const testDatabase: MongoTestDatabase = new MongoTestDatabase()
+	beforeEach(async () => await testDatabase.setupDatabase())
+	afterEach(async () => await testDatabase.teardownDatabase())
```

Comment:
Which one are you pointing at?
I suspect the one for teardown, but i am not 100% sure.

Reply:
I am pointing at both.
The type of the lambda functions are currently `() => Promise<void>`.
If you remove the async/await from both, you get the exact same type. The async/await syntax would only be useful for the lambdas contained multiple promise-based calls that should be executed in sequence.

Reply:
Async cannot be removed due to our linting rules, as the called methods (`setupDatabase`/`teardownDatabase`) both return a promise.

Maybe a small chat about this could help.

Reply:
`await` has been removed, as per aggrement during chat.